zcash_client_sqlite/src/wallet/init/migrations/transparent_gap_limit_handling.rs

```diff
@@ -0,0 +1,556 @@
+                let account_id: i64 = row.get("account_id")?;
+                account_ids.insert(account_id);
+
+                let uivk = decode_uivk(row.get("uivk")?)?;
+                let di_be: Vec<u8> = row.get("diversifier_index_be")?;
+                let diversifier_index = decode_diversifier_index_be(&di_be)?;
+
+                let transparent_external = NonHardenedChildIndex::try_from(diversifier_index)
+                    .ok()
+                    .and_then(|idx| {
+                        uivk.transparent()
+                            .as_ref()
+                            .and_then(|external_ivk| external_ivk.derive_address(idx).ok())
+                            .map(|t_addr| (idx, t_addr))
+                    });
+
+                // Add transparent address index metadata and the transparent address corresponding
+                // to the index to the addresses table. We unconditionally set the cached
+                // transparent receiver address in order to simplify gap limit handling; even if a
+                // unified address is generated without a transparent receiver, we still assume
+                // that a transparent-only wallet for which we have imported the seed may have
+                // generated an address at that index.
+                if let Some((idx, t_addr)) = transparent_external {
+                    conn.execute(
+                        r#"
+                        UPDATE addresses
+                        SET transparent_child_index = :transparent_child_index,
+                            cached_transparent_receiver_address = :t_addr
+                        WHERE account_id = :account_id
+                        AND diversifier_index_be = :diversifier_index_be
+                        "#,
+                        named_params! {
+                            ":account_id": account_id,
+                            ":diversifier_index_be": &di_be[..],
+                            ":transparent_child_index": idx.index(),
+                            ":t_addr": t_addr.encode(&self.params),
+                        },
+                    )?;
+                }
+            }
+        }
+
+        // We now have to re-create the `addresses` table in order to fix the constraints. Note
+        // that we do not include the `seen_in_tx` column as this is duplicative of information
+        // that can be discovered via joins with the various `*_received_{notes|outputs}` tables,
+        // which we will create a view to perform below. The `used_in_tx` column data is used only
+        // to determine the height at which the address was exposed (for which we use the target
+        // height for the transaction.)
+        conn.execute_batch(&format!(
+            r#"
+            CREATE TABLE addresses_new (
+                id INTEGER NOT NULL PRIMARY KEY,
+                account_id INTEGER NOT NULL,
+                key_scope INTEGER NOT NULL DEFAULT {external_scope_code},
```

Comment:
Do we need to leave the default set? It's convenient for the migration (so could be retained above), but it could cause us issues during future query refactoring if the final table auto-fills the column (fixing a data error here would not be pleasant).

Reply:
Hm, I think you're correct that we may not need the default, I will check.

Reply:
Fixed in c3e9b06fa3eef401aacafdca1dae609829bfe1d6